packages/sfui/frameworks/react/hooks/useScrollable/useScrollable.ts

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useRef, useState } from 'react';
+import { type UseScrollableOptions, Scrollable, composeHandlers } from '@storefront-ui/react';
+
+export function useScrollable<TElement extends HTMLElement>({
+  activeIndex,
+  direction,
+  drag,
+  reduceMotion,
+  onDragChange,
+  onScroll,
+  onPrev,
+  onNext,
+}: Partial<UseScrollableOptions> = {}) {
+  const containerElement = useRef<TElement>(null);
+  const scrollable = useRef<Scrollable | null>(null);
+  const [state, setState] = useState({ hasPrev: false, hasNext: false, isDragged: false });
+
+  useEffect(() => {
+    const container = containerElement.current;
+    if (!container) {
+      return () => {};
+    }
+    scrollable.current = new Scrollable(container, {
+      direction,
+      drag,
+      reduceMotion,
+      onPrev,
+      onNext,
+      onScroll: (data) => {
+        setState((currentState) => ({ ...currentState, hasNext: data.hasNext, hasPrev: data.hasPrev }));
+        onScroll?.(data);
+      },
+      onDragChange: (data) => {
+        setState((currentState) => ({ ...currentState, isDragged: data.isDragged }));
+        onDragChange?.(data);
+      },
+    });
+    const unregister = scrollable.current.register();
+
+    if (activeIndex) scrollable.current.scrollToIndex(activeIndex);
+
+    return unregister;
+  }, [containerElement, activeIndex, direction, drag, reduceMotion, onDragChange, onScroll, onPrev, onNext]);
+
+  const getPrevButtonProps = <TProps extends { onClick: () => void }>(props?: TProps) => {
```

Comment:
Correct me if I'm wrong, but I believe we could use `React.HTMLProps` here. For sure, if we keep it like this, it would expect to always pass objects with onClick. I can recall we already have a helper function `createPropsGetter`. Could you tray to use it here? You can see usage example in `useTooltip`.